apps/site/pages/ltb/entry/[slug]/index.tsx

```diff
@@ -0,0 +1,1041 @@
+import {
+	getLearningDiaryInformation,
+	LearningDiaryEntryResult,
+	LearningDiaryInformation,
+	LearningStrategy,
+	LearningTechnique,
+	LearningTechniqueEvaluation
+} from "@self-learning/api";
+import { useRouter } from "next/router";
+
+export default function LearningDiaryEntryOverview({
+	learningDiaryInformation,
+	slug
+}: {
+	learningDiaryInformation: LearningDiaryInformation;
+	slug: string | null;
+}) {
+	const router = useRouter();
+
+	const [currentIndex, setCurrentIndex] = useState<number>(() => {
+		const initialIndex =
+			slug && !isNaN(Number(slug))
+				? Number(slug) - 1
+				: learningDiaryInformation?.learningDiaryEntries?.length
+				? learningDiaryInformation.learningDiaryEntries.length - 1
+				: 0;
+		return initialIndex < 0 ? 0 : initialIndex;
+	});
+
+	const [entries, setEntries] = useState<LearningDiaryEntryResult[]>(
+		learningDiaryInformation.learningDiaryEntries
+	);
+
+	const handleUpdateEntry = (updatedEntry: LearningDiaryEntryResult) => {
+		setEntries(prevEntries =>
+			prevEntries.map(entry => (entry.id === updatedEntry.id ? updatedEntry : entry))
+		);
+	};
+
+	useEffect(() => {
+		window.history.pushState({}, "", "/ltb/entry/" + (currentIndex + 1));
+	}, [currentIndex]);
+
+	return (
+		<div className="flex justify-center">
+			<div className="w-2/3 py-4">
+				<div className="mb-4 flex justify-center">
+					<EntrySwitcher
+						maxLength={entries.length}
+						setIndex={setCurrentIndex}
+						currentIndex={currentIndex}
+					/>
+				</div>
+				<LearningDiaryEntryForm
+					entry={entries[currentIndex]}
+					learningDiaryInformation={learningDiaryInformation}
+					onUpdate={handleUpdateEntry}
+				/>
+			</div>
+		</div>
+	);
+}
+
+function EntrySwitcher({
+	maxLength,
+	setIndex,
+	currentIndex
+}: {
+	maxLength: number;
+	setIndex: (index: number) => void;
+	currentIndex: number;
+}) {
+	const handlePrev = () => {
+		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+		// @ts-expect-error
+		setIndex(prevIndex => Math.max(prevIndex - 1, 0));
+	};
+
+	const handleNext = () => {
+		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+		// @ts-expect-error
+		setIndex(prevIndex => Math.min(prevIndex + 1, maxLength - 1));
+	};
+
+	return (
+		<div className="flex space-x-16">
+			<button className="btn btn-primary" onClick={handlePrev} disabled={currentIndex === 0}>
+				Vorheriger Eintrag
+			</button>
+			<div className="rounded bg-gray-100 p-2">{currentIndex + 1}</div>
+			<button
+				className="btn btn-primary"
+				onClick={handleNext}
+				disabled={currentIndex === maxLength - 1}
+			>
+				Nächster Eintrag
+			</button>
+		</div>
+	);
+}
+
+function LearningDiaryEntryForm({
+	entry,
+	learningDiaryInformation,
+	onUpdate
+}: {
+	entry: LearningDiaryEntryResult;
+	learningDiaryInformation: LearningDiaryInformation;
+	onUpdate: (updatedEntry: LearningDiaryEntryResult) => void;
+}) {
+	const methods = useForm({
```

Comment:
variablen name ungünstig gewählt. `form` ist da schon etwas klarer und wird bisher in der restlichen anwendung verwendet